hw/ip/pwm/doc/_index.md

```diff
@@ -0,0 +1,181 @@
+- a single PWM enable bit (e.g. {{< regref "PWM_EN.EN_0" >}})
+- a 16-bit PWM duty cycle (e.g. {{< regref "PWM_PARAM_0.PHASE_DELAY_0" >}})
+- a 16-bit PWM phase delay ({{< regref "PWM_PARAM_0.DUTY_CYCLE_0" >}})
+- a single blink enable bit ({{< regref "BLINK_EN.EN_0" >}})
+- a 16-bit blink rate divider (e.g. {{< regref "BLINK_PARAM_0.RATE_0" >}})
+- an 8-bit blink duty cycle (e.g. {{< regref "BLINK_PARAM_0.DELAY_0" >}})
+- an 8-bit blink phase adjust (e.g. {{< regref "BLINK_PARAM_0.DUTY_CYCLE_0" >}})
+
+In the operational description below, all of the examples discuss the operation of channel 0, using registers and fields ending with "_0".
+To operated other channels, simply choose the registers with the appropriate channel suffix.
+
+Each PWM FSM output stays low when disabled, by clearing {{< regref PWM_EN.EN_0 >}}.
+
+The channel output rises when the phase counter equals the corresponding {{<regref "PWM_PARAM_0.PHASE_DELAY_0">}}, and falls when the phase counter equals {{< regref "PWM_PARAM_0.PHASE_DELAY_0">}} + {{< regref "PWM_PARAM_0.DUTY_CYCLE_0" >}} + 1 (mod 65536).
+The following figure illustrates the effect of the {{< regref "PWM_PARAM_0.PHASE_DELAY_0" >}} and {{< regref "PWM_PARAM_0.DUTY_CYCLE_0" >}} register fields.
+Note that this figure shows two channels, 0 and 1, where the second channel has a significant phase delay, such that the output pulse is high when `phase_ctr` overflows to zero.
+
+{{< wavejson >}}
+{signal: [
+  {name: 'PWM_EN', wave: '01...|....|......'},
+  {name: 'phase_ctr', wave: '2.222|2222|222222', data: ['0', '1', '2', '', '1024', '1025', '1026', '', '65533', '65534' ,'65535', '0', '1', '2', '3', '4']},
+  {name: 'PHASE_DELAY_0', wave: '2....|....|......', data: '0'},
+  {name: 'DUTY_CYCLE_0', wave: '2....|....|......', data: '1023'},
+  {name: 'pwm_out[0]', wave: '0.1..|.0..|....1.'},
+  {name: 'PHASE_DELAY_1', wave: '2....|....|......', data: '65533'},
+  {name: 'DUTY_CYCLE_1', wave: '2....|....|......', data: '3'},
+  {name: 'pwm_out[1]', wave: '01.0.|....|.1...0'}
+  ],
+ config:{skin:'narrow', hscale:2}
+}
+{{< /wavejson >}}
+
+#### Hardware-Controlled Blink Feature
+
+In addition, to the PWM enable function there is also a hardware-controlled blink feature to periodically hold the output low for many PWM cycles.
+The blink feature creates a blink-control signal, which modulates the PWM output.
+If hardware-controlled blinking is enabled, the PWM pulses are suppressed whenever the blink-control signal is low.
+
+The blink rate can be controlled to be as fast as once every 2 PWM cycles or as slow as once every 65536 PWM cycles.
+This rate is controlled by the combination of two registers {{< regref "BLINK_PARAM_0.RATE_0" >}} and {{< regref "BLINK_PARAM_0.PREDIV_0" >}}.
+The {{< regref "BLINK_PARAM_0.RATE_0" >}} register determines the period of the blink-control signal was well as the resolution in blink phase and duty-cycle, with a minimum phase resolution ranging from 1/256 to 1/2 of a cycle.
```

Comment:
was well -> as well

Reply:
Thanks, will fix.

Reply:
(This sentence has been completely rewritten)